### ISSUE TITLE
Unable to run tasks in sequence

### ISSUE THREAD

**Issue by lightvision** [laravel/elixir#180]
I know that task are done in parallel, but with gulp I can run tasks in a predefined order and make a task to wait completion of another task.

If I use only gulp I loose the beauty of elixir. If I use only elixir, all tasks with the same name will run syncrounously, but I depend on some tasks to be finished before others (compiling coffe, merging the resulting js with another one the merging that js with previouly builded one).

If I use ```elixir.extend()...``` I can define tasks that runs in sequence, but that extended tasks don't produce the desired output.

So, whats the cleaner solution to achieve this?

Note: running gulp when I have only the following code in ``gulpfile.js``:
```
elixir(function (mix) {
    
    
});
```

gave me the following output:
```
[12:32:34] Using gulpfile ~/projects/hdcrmwithdingoapi.dev/gulpfile.js
[12:32:34] Starting 'default'...
[12:32:34] 'default' errored after 28 ms
[12:32:34] Error: No tasks were provided to run-sequence
    at verifyTaskSets (/home/vagrant/projects/app.dev/node_modules/laravel-elixir/node_modules/run-sequence/index.js:9:9)
    at runSequence (/home/vagrant/projects/app.dev/node_modules/laravel-elixir/node_modules/run-sequence/index.js:85:2)
    at Gulp.<anonymous> (/home/vagrant/projects/app.dev/node_modules/laravel-elixir/ingredients/default.js:17:16)
    at module.exports (/home/vagrant/projects/app.dev/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/home/vagrant/projects/app.dev/node_modules/gulp/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/home/vagrant/projects/app.dev/node_modules/gulp/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/home/vagrant/projects/app.dev/node_modules/gulp/node_modules/orchestrator/index.js:134:8)
    at /usr/lib/node_modules/gulp/bin/gulp.js:129:20
    at process._tickCallback (node.js:355:11)
    at Function.Module.runMain (module.js:503:11)
```

so definitively must be a way to run some task in sequence and other in parallel.

**Comment by JeffreyWay** [comment]
I may rewrite some of this for the next big release of Elixir, but right now, that's how it works. They all get added to an array. So your mix.sass() calls will be grouped together.

**Comment by x-yuri** [comment]
But can you please suggest any workaround? Or idea for workaround? How are you going to fix this?

**Comment by parsnick** [comment]
Tasks have run in sequence since [version 3.0](https://github.com/laravel/elixir/releases/tag/3.0.0).

**Comment by x-yuri** [comment]
I can see [the opposite][1].

[1]: http://stackoverflow.com/questions/34235447/how-to-run-two-elixir-tasks-in-sequence

**Comment by parsnick** [comment]
Probably an issue with your custom tasks rather than Elixir. 

Looks like you're not giving gulp any way of knowing when your tasks are finished - try returning the stream.

**Comment by x-yuri** [comment]
Already did, to no avail. Can you show me in the code what makes the tasks run in sequence?

**Comment by parsnick** [comment]
I guess it's the use of `run-sequence` in [tasks/default:18](https://github.com/laravel/elixir/blob/master/tasks/default.js#L18).

Sounds like it might be worth opening a new issue then? This was closed because it was known to be unsupported at the time.
